### PR TITLE
[#2998] Allow both release and develop version to coexist

### LINKED ISSUE #2998: Allow both release and develop version to coexist
### User Story

As a developer, I want to have both release and developer version running at the same time on android.

### Description

Currently both versions share same identifier preventing to keep the released version installed.

*Type*: Feature

### PR DESCRIPTION
**Pull request by endenwer**
fixes #2998

### Summary:
Change app id for debug build type so both degub and release build type can coexist. If build debug version with dev clojure profile(which is default profile for dev) and release version with prod clojure profile(witch is also default) then function `status.im.core.debug?` will return `true` for degub build and `false` for release build.

### Steps to test:
- Install release version
- Install debug version
- Check that two apps exist: Status and Status Debug.

photo_2018-04-26_13-30-22
photo_2018-04-26_13-30-18

status: ready

### CODE REVIEW

File: src/status_im/core.cljs [hunk 1]
```diff
@@ -8,7 +8,10 @@
             [status-im.react-native.js-dependencies :as js-dependencies]
             [goog.object :as object]))
 
-(when js/goog.DEBUG
+(defonce debug?
```
Comment (jeluard) [inline src/status_im/core.cljs]: What does this provide us?

**Review by jeluard — COMMENTED** [review]
Is there a way to distinguish both versions before launching them?
Also in the app itself it would be nice to know it's the debug version (e.g. some label in user profile)

**Comment by jeluard** [comment]
@endenwer Thanks for looking at this!

File: src/status_im/core.cljs (hunk 1 above)
Comment (endenwer) [reply src/status_im/core.cljs]: A way to know which version is running. But it only shows cljs build, not android build.

**Comment by endenwer** [comment]
I can use this https://github.com/APSL/react-native-version-number as in debug build there is `-SNAPSHOT` suffix in version and just by looking at version I can say if it is debug build or not. But I need to make it work on ios.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/APSL/react-native-version-number">APSL/react-native-version-number</a></strong></div><div>react-native-version-number - Gets the version number and build number of your app.</div></blockquote>

File: src/status_im/core.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/core.cljs]: As opposed to using `js/goog.DEBUG` directly?

Comment (endenwer) [reply src/status_im/core.cljs]: just convenience :) Or it is not necessary?

**Comment by jeluard** [comment]
Version is already available here: https://github.com/status-im/status-react/blob/develop/src/status_im/utils/build.cljs

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>status-react - a free (libre) open source, mobile OS for Ethereum</div></blockquote>

File: src/status_im/core.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/core.cljs]: I'd rather stick with the default convention

Comment (endenwer) [reply src/status_im/core.cljs]: ok, I remove it then.

**Comment by endenwer** [comment]
> Is there a way to distinguish both versions before launching them?

They have different names: Status and Status debug.

**Comment by endenwer** [comment]
> Also in the app itself it would be nice to know it's the debug version (e.g. some label in user profile)

@jeluard I will add it. Do you have any specific place in mind where I can place it?

**Comment by jeluard** [comment]
@endenwer In the profile maybe?
@denis-sharypin Can you guide us here? How can we distinguish debug builds from release build from the app itself? Or should we do it at all?

**Comment by endenwer** [comment]
@jeluard ok, I will add centered `debug` label at the end of profile screen. Maybe make it as `Advanced` button but without arrow and another color.

**Comment by endenwer** [comment]
For now I can you `js/goog.DEBUG` to distinguish debug build. It is the easiest way and will work on both android and ios.

**Comment by jeluard** [comment]
@endenwer BTW do you know if this can be done for iOS too? We could create a bounty for it.

**Comment by endenwer** [comment]
@jeluard Yes, it can be done for ios too. And also I can make different icon for debug version but I need someone to make the icon.

**Comment by jeluard** [comment]
@endenwer Cool! Will create a bounty then.
@denis-sharypin Can we have an icon for debug builds please? :)

**Comment by jeluard** [comment]
@endenwer When you have spare time you can look at this too (iOS): https://github.com/status-im/status-react/issues/4010 :)

**Comment by denis-sharypin** [comment]
@jeluard @endenwer 
I can create a separate icon ofc,
pls create an issue and mention me there

**Comment by endenwer** [comment]
debug build fails with:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
        at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
        at org.json.JSONTokener.nextValue(JSONTokener.java:94)
        at org.json.JSONObject.<init>(JSONObject.java:156)
        at org.json.JSONObject.<init>(JSONObject.java:173)
        at im.status.ethereum.module.StatusModule.doStartNode(StatusModule.java:212)
        at im.status.ethereum.module.StatusModule.access$000(StatusModule.java:32)
        at im.status.ethereum.module.StatusModule$1.run(StatusModule.java:362)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
```
This is because `network` key in db is nil. The key is nil because its default value is loaded by `js-dependencies/config` which is `(js/require "react-native-config")`. So problem with `js/require` or with loading react native dependencies. 

Will try to fix this tomorrow but I have no idea what is the problem.

**Comment by jeluard** [comment]
@endenwer Thanks!
@denis-sharypin See https://github.com/status-im/status-react/issues/4018

**Comment by endenwer** [comment]
@jeluard Fixed it. Profile page in debug build:
photo_2018-04-26_22-09-15
Not sure about design. I am not a designer 😀 
Wait for the icon.

**Comment by jeluard** [comment]
@endenwer Thanks! All things considered we will not show anything in the app, just have a different icon.. Sorry about that.
Let's wait for the new icon from @denis-sharypin

**Comment by endenwer** [comment]
ok, I will remove that.

**Comment by endenwer** [comment]
@jeluard Done.
s80429-103325

**Review by jeluard — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Comment by asemiankevich** [comment]
So i was about to test it and installed the PR android build first and this does not allow me to install release in parallel, as well as the PR build icon is not marked as `debug`. I talked with @jeluard and he said we need to make an agreement on what to consider a debug build first (nightly / PR) and then configure our Jenkins if needed. 
Not sure what to do next before this is resolved.

**Comment by endenwer** [comment]
@asemiankevich How do you build debug build?

**Comment by asemiankevich** [comment]
@endenwer well as i mentioned above i did not do that. I used the PR build (made by Jenkins).

**Comment by endenwer** [comment]
Jenkins create release builds.

**Comment by asemiankevich** [comment]
yes and i can't build it locally unfortunately

**Review by goranjovic — APPROVED** [review]

**Comment by jeluard** [comment]
@asemiankevich How can we move this forward?
@endenwer Can you provide a link to the android build to help @asemiankevich

**Comment by endenwer** [comment]
@jeluard @asemiankevich Debug apk https://www.dropbox.com/s/2qi761j05qf3y1h/app-debug.apk

Build commands:
```
make prepare prod-build-android
mkdir android/app/src/main/assets
node --expose-gc --max_old_space_size=4096 ./node_modules/react-native/local-cli/cli.jsbundle --platform android --dev false --entry-file index.android.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/debug/res
make run-android
```




<blockquote><div>Dropbox</div><div><strong><a href="https://www.dropbox.com/s/2qi761j05qf3y1h/app-debug.apk">app-debug.apk</a></strong></div><div>Shared with Dropbox</div></blockquote>

**Comment by jeluard** [comment]
@endenwer Thanks!
@asemiankevich Please try this one :)

**Comment by asemiankevich** [comment]
@endenwer @jeluard i have downloaded the attached apk but it is not possible to install it for some reason. Anyway i dont want to hold this change . I will build this debug build later today by myself and play around. Please merge it

**Pull request merged**